### ISSUE TITLE
Changing difficulty from peaceful to easy in Skript needs a server restart

### ISSUE THREAD

**Issue by oskarkk** [SkriptLang/Skript#3834]
### Description
I have a script that allows the players to vote on difficulty changes and it sometimes doesn't work. I believe something is broken in Skript.

### Steps to Reproduce
1. Start a new server with `difficulty=peaceful` in server.properties, or run `/difficulty peaceful` on an already running server and stop the server, so that the test world would be set to peaceful.
2. Write a script containing:

```
command /test <text>:
  trigger:
    set difficulty of world "world" to arg-1 parsed as difficulty
```

or:

```
    set difficulty of all worlds to arg-1 parsed as difficulty
```

3. Load the script, log into the server, run `/time set night` and then `/test easy`
4. Nothing happens, no mobs appear. `/difficulty easy` says that the world is already set to easy.
5. Restart the server.
6. Log in, the mobs spawn as it's easy.

And the same scenario without using Skript:

1. Start a server with worlds set to peaceful as above
2. Run `/difficulty easy`.
3. Mobs spawn instantly.

And then:

4. Run `/difficulty peaceful`.
5. `/test easy` - again, nothing happens.
5. `/test peaceful`  and `/difficulty easy` - mobs spawn
6. `/test peaceful` and `/test easy` - this time, when we first changed easy to peaceful in Skript, mobs spawn on change from peaceful to easy
7. Restart the server, the difficulty is easy, mobs spawn normally
8. `/test peaceful` - mobs disappear.
9. `/test easy` - mobs appear

It seems that there's a bug only on the initial change from peaceful to easy, strange thing.

### Expected Behavior
The same as the `/difficulty` command

### Server Information
* **Server version/platform:** This server is running Paper version git-Paper-543 (MC: 1.16.5) (Implementing API version 1.16.5-R0.1-SNAPSHOT)
* **Skript version:** Skript version 2.5.3
* System Info: Java 15 (OpenJDK 64-Bit Server VM 15.0.2+7) Host: Linux 5.10.19-1-MANJARO (amd64)

**Comment by TPGamesNL** [comment]
Can confirm

**Comment by oskarkk** [comment]
Fixed in #3921, thanks

**Comment by TPGamesNL** [comment]
We like to keep issues open with the completed label until they appear in a release :)
